todo_app/app.py

```diff
@@ -11,18 +12,74 @@
 from .data.view_model import *
 
 
+
+# Define the User class
+class User(UserMixin):
+    def __init__(self, id):
+        self.id = id
+
 def create_app():
     app = Flask(__name__)
     app.config.from_object(Config())
 
+    # Set LOGIN_DISABLED based on the environment variable
+    app.config['LOGIN_DISABLED'] = os.getenv('LOGIN_DISABLED') == 'True'
+
+    login_manager = LoginManager()
+
+    @login_manager.unauthorized_handler
+    def unauthenticated():
+        github_auth_url = construct_github_url()
+        return redirect(github_auth_url)
+    
+    @login_manager.user_loader
+    def load_user(user_id):
+        return User(user_id)
+
+    login_manager.init_app(app)
+
+
+    @app.route('/.auth/login/github/callback')
+    def github_callback():
+        # Extract the 'code' from the query parameters
+        code = request.args.get('code')
+        if not code:
+            return 'Authorization request failed', 400
+
+        # Exchange the code for an access token
+        access_token = exchange_code_for_token(code)
+
+        
+        if not access_token:
+            return 'Failed to fetch access token', 400
+        
+        #Get the authenticated user
+        authenticated_user = get_github_user(access_token)
+        
+
+        if not authenticated_user:
+            return 'Failed to fetch the authenticated user', 400
+        
+        # Construct a new instance of the User class
+        user = User(authenticated_user['id'])
+        login_user(user)
+
+        # Store the access token in a session or a secure place
+        session['access_token'] = access_token
```

Comment:
Is there any reason to actually store the token here? Once the user is logged in, simply setting some appropriate user_info should be sufficient here, and reduces the risk that we leak the token somewhere!

Reply:
I wish to prevent users from having to log in every time they access the app. Instead, if they possess a valid token within their session, the app should utilize that. Could you advise on how to implement this securely

Reply:
First, how does that differ from the default behaviour? I believe currently they will get redirected to GitHub, _but_ it will silently approve them if it's seen them before, and they'll be redirected back, so there won't be a conscious log-in-again process. That said, if you want to avoid that redirect, then really we want to focus on the Flask session lasting longer so it doesn't need to re-authenticate with GitHub; I haven't had a go but you could look at [the Remember Me functionality](https://flask-login.readthedocs.io/en/latest/#remember-me) Flask provides